adagrams/game.py

```diff
@@ -1,13 +1,197 @@
+                   'G':2,
+                   'H':4,
+                   'I':1,
+                   'J':8,
+                   'K':5,
+                   'L':1,
+                   'M':3,
+                   'N':1,
+                   'O':1,
+                   'P':3,
+                   'Q':10,
+                   'R':1,
+                   'S':1,
+                   'T':1,
+                   'U':1,
+                   'V':4,
+                   'W':4,
+                   'X':8,
+                   'Y':4,
+                   'Z':10
+                   }
+
+
+def make_dict(array):
+    '''
+    Creates a dictionary of letters in the array as keys, and frequency of letters as values 
+    input: a sting or a list of characters
+    output: a dictionary of letters as keys, and frequency as values
+    '''
+    make_dict = {}
+    for character in array:
+        if character not in make_dict:
+            make_dict[character] = 1
+        else:
+            make_dict[character] += 1
+    return make_dict
+
+
 def draw_letters():
-    pass
+    # Convert letter pool dictionary to a list with repeated letters 
+    # the list above has 98 letters.
+    # make a random selection out of 98 letters of pool list.
+    # add the selected letter to the hand and add a counter that shows frequency of the letter in the hand
+    # draw hand should be a dict array with k/v as letters / freq
+    # check if freq of the selected letter is less than the freq of the pool
+    # repeat this 10 times!
+    # return a list of hand letters!
+    
+    letter_pool_list = []
+    hand_dict = {}
+    hand_list = []
+
+    for letter, frequency in LETTER_POOL.items():
+        for i in range(frequency):
+            letter_pool_list.append(letter)
+
+    # for i in range(10): Can't use for loop in indetermined number of loops
+    hand_counter = 1
+    while hand_counter <= 10:
+        draw = randint(0, 97)
+        letter = letter_pool_list[draw]
+
+        if letter not in hand_dict:
+            hand_dict[letter] = 1
+        else:
+            if hand_dict[letter] < LETTER_POOL[letter]:
+                hand_dict[letter] += 1
+            else:
+                continue
+        
+        hand_counter += 1
+
+    for letter, frequency in hand_dict.items():
+        for i in range(frequency):
+            hand_list.append(letter)
+
+    return hand_list
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    # first need to understand what are letters in word and how many of the letter do we have.
+    # then I need to compare each letter with my draw bank:
+    # if the letter in the word doesn't exist in the draw return false
+    # if the letter is in the draw bank then check if the frequency in the word 
+    # is more than the frequency of same letter in the draw bank
+    
+    word = word.upper()
+    word_dict = make_dict(word)
+    letter_bank_dict = make_dict(letter_bank)
+
+    for character, frequency in word_dict.items():
```

Comment:
One of the main reasons we use dictionaries is that they run lookup in constant time. Looping through them is sometimes necessary, but it does defeat the purpose of the constant lookup. If we can avoid looping through a dictionary, we should definitely try to do so! This would work the exact same way if we looped through our input word!